### PR TITLE
[gawb-3964] automation test to check sam policies and acls in google

### PR DESCRIPTION
**Pull request by marctalbott**
[GAWB-3964](https://broadinstitute.atlassian.net/browse/GAWB-3964)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala [hunk 1]
```diff
@@ -61,7 +61,7 @@ class AuthDomainSpec extends FlatSpec with Matchers with CleanUp with BillingFix
     withGroup("ad", List(projectUser.email, projectOwner.email)) { realmGroup =>
       withGroup("ad2", List(projectUser.email, projectOwner.email)) { realmGroup2 =>
         val realmGroup2Full = Orchestration.groups.getGroup(realmGroup2)(groupOwnerToken)
-        withGroup("ad3", List(realmGroup2Full.groupEmail)) { realmGroup3 =>
+        withGroup("ad3", List(realmGroup2Full.membersGroup.groupEmail)) { realmGroup3 =>
```
Comment (dvoet) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala:64]: why the changes in this file?

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 2]
```diff
@@ -351,5 +353,100 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
...
+            case acl if (acl.getRole() != null && !acl.getRole().equals("OWNER")) => (acl.getRole(), acl.getEmail())
+          }
+          val expectedObjectRolesAndPolicies = unconstrainedSamPolicies.collect {
+            case AccessPolicyResponseEntry("project-owner", AccessPolicyMembership(emails, _, _), _) => ("READER", emails.head)
+            case AccessPolicyResponseEntry(policyName, _, email) if policyToObjectAccessLevel.contains(policyName) => (policyToObjectAccessLevel(policyName), email.value)
+          }
+          eventually(actualObjectRolesAndPolicies should contain theSameElementsAs expectedObjectRolesAndPolicies)
+        }
+
+        withGroup("authDomain", List(owner.email)) { authDomain =>
```
Comment (dvoet) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala:424]: I would make this a separate test

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 3]
```diff
@@ -351,5 +353,100 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
...
+      val serviceAccountEmailDomain = "developer.gserviceaccount.com"
+
+      withCleanBillingProject(owner) { projectName =>
+        withWorkspace(projectName, s"unconstrained-workspace") { workspaceName =>
+          val workspaces = Rawls.workspaces.list()
+          val workspaceId = workspaces.parseJson.asInstanceOf[JsArray].elements.flatMap { workspaceListEntry =>
+            workspaceListEntry.asJsObject.getFields("workspace").flatMap { workspace =>
+              workspace.asJsObject.getFields("workspaceId")
+            }
+          }.head.convertTo[String]
```
Comment (dvoet) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala:377]: there is no guarantee that the first workspace is the one you are looking for... find the workspace in the list with the right namespace (should match projectName) and name (should match workspaceName).

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala (hunk 1 above)
Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala:64]: There was a change to the `OrchestrationModel` at some point and `.groupEmail` is no longer a part of the `ManagedGroupWithMembers` class, it was moved to the `RawlsGroupShort` class. This change was needed for Rawls to compile when I updated the dependency to use a newer version of `serviceTest`. I'm now realizing that I had only updated the dependency for debugging purposes, so I can remove this change when I revert the `serviceTest` version to the older version.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 4]
```diff
@@ -351,5 +353,100 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
...
+
+          // check sam policies
+          val unconstrainedSamPolicies = Sam.user.listResourcePolicies("workspace", workspaceId)
+          val unconstrainedSamPolicyNames = unconstrainedSamPolicies.map(_.policyName)
+          eventually(unconstrainedSamPolicyNames should contain theSameElementsAs workspacePolicyNames)
```
Comment (dvoet) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: don't need eventually

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala (hunk 1 above)
Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala:64]: Actually I do need to update the `serviceTest` version, so I think I'll need to keep these changes.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 5]
```diff
@@ -351,5 +353,100 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
...
+          // service account should have owner access
+          assert(unconstrainedBucketAcl.exists(acl => acl.getRole().equals("OWNER") && acl.getEmail().endsWith(serviceAccountEmailDomain)))
+          val actualBucketRolesAndPolicies = unconstrainedBucketAcl.collect {
+            case acl if (acl.getRole() != null && !acl.getRole().equals("OWNER")) => (acl.getRole(), acl.getEmail())
+          }
+          val expectedBucketRolesAndPolicies = unconstrainedSamPolicies.collect {
+            case AccessPolicyResponseEntry("project-owner", AccessPolicyMembership(emails, _, _), _) => ("WRITER", emails.head)
+            case AccessPolicyResponseEntry(policyName, _, email) if policyToBucketAccessLevel.contains(policyName) => (policyToBucketAccessLevel(policyName), email.value)
+          }
+          eventually(actualBucketRolesAndPolicies should contain theSameElementsAs expectedBucketRolesAndPolicies)
```
Comment (dvoet) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: no eventually

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala (hunk 1 above)
Comment (dvoet) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala:64]: something doesn't seem right because we should not be using the membersGroup email for anything

Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainSpec.scala:64]: Okay, what email should be used then? The `ManagedGroupWithMembers` class doesn't have a group email associated with it anymore, and I am not sure what it should be replaced with.
